### PR TITLE
Add Group* endpoints

### PR DESCRIPTION
**Pull request by janinko**

### CODE REVIEW

**Comment by janinko** [comment]
retest this please

File: rest-new/src/main/java/org/jboss/pnc/rest/endpoints/GroupBuildEndpointImpl.java [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+import org.slf4j.LoggerFactory;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.ws.rs.NotFoundException;
+
+/**
+ *
+ * @author Honza Brázdil &lt;jbrazdil@redhat.com&gt;
+ */
+@ApplicationScoped
```
Comment (thescouser89) [inline rest-new/src/main/java/org/jboss/pnc/rest/endpoints/GroupBuildEndpointImpl.java]: `@Stateless` maybe?

File: rest-new/src/main/java/org/jboss/pnc/rest/endpoints/GroupConfigurationEndpointImpl.java [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.enterprise.context.ApplicationScoped;
+
+/**
+ *
+ * @author Honza Brázdil &lt;jbrazdil@redhat.com&gt;
+ */
+@ApplicationScoped
```
Comment (thescouser89) [inline rest-new/src/main/java/org/jboss/pnc/rest/endpoints/GroupConfigurationEndpointImpl.java]: @Stateless maybe?

File: rest-new/src/main/java/org/jboss/pnc/rest/endpoints/BuildEndpointImpl.java [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+import org.jboss.pnc.rest.api.parameters.BuildsFilterParameters;
+import org.jboss.pnc.rest.api.parameters.PageParameters;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ *
+ * @author Honza Brázdil &lt;jbrazdil@redhat.com&gt;
+ */
+public class BuildEndpointImpl extends AbstractEndpoint<Build, BuildRef> implements BuildEndpoint {
```
Comment (thescouser89) [inline rest-new/src/main/java/org/jboss/pnc/rest/endpoints/BuildEndpointImpl.java]: Missing EJB annotation!

File: rest-new/src/main/java/org/jboss/pnc/rest/endpoints/GroupBuildEndpointImpl.java (hunk 1 above)
Comment (janinko) [reply rest-new/src/main/java/org/jboss/pnc/rest/endpoints/GroupBuildEndpointImpl.java]: Why? It doesn't have to be in transactions.

File: rest-new/src/main/java/org/jboss/pnc/rest/endpoints/BuildEndpointImpl.java (hunk 3 above)
Comment (janinko) [reply rest-new/src/main/java/org/jboss/pnc/rest/endpoints/BuildEndpointImpl.java]: Yeah, missing annotation but why EJB and not CDI?

Also from [Resteasy doc](https://docs.jboss.org/resteasy/docs/3.6.1.Final/userguide/html/CDI.html):
> If a JAX-RS root resource does not define a scope explicitly, it is bound to the Request scope.

Comment (thescouser89) [reply rest-new/src/main/java/org/jboss/pnc/rest/endpoints/BuildEndpointImpl.java]: Sure go with CDI then!

File: rest-new/src/main/java/org/jboss/pnc/rest/endpoints/GroupBuildEndpointImpl.java (hunk 1 above)
Comment (thescouser89) [reply rest-new/src/main/java/org/jboss/pnc/rest/endpoints/GroupBuildEndpointImpl.java]: Wouldn't @ApplicationScoped create a singleton basically?

File: rest-new/src/main/java/org/jboss/pnc/rest/endpoints/BuildEndpointImpl.java (hunk 3 above)
Comment (thescouser89) [reply rest-new/src/main/java/org/jboss/pnc/rest/endpoints/BuildEndpointImpl.java]: We'll have to sync all the other EndpointImpl to use the same CDI annotation/EJB though

File: rest-new/src/main/java/org/jboss/pnc/rest/endpoints/GroupBuildEndpointImpl.java (hunk 1 above)
Comment (janinko) [reply rest-new/src/main/java/org/jboss/pnc/rest/endpoints/GroupBuildEndpointImpl.java]: Yes it would. That's why I'm using it.

Comment (thescouser89) [reply rest-new/src/main/java/org/jboss/pnc/rest/endpoints/GroupBuildEndpointImpl.java]: Does that affect the throughput when requests go in? If I remember correctly, its 1 thread per request? But yeah I dunno how this behaves with CDI and no concurrency limit.

File: rest-new/src/main/java/org/jboss/pnc/rest/endpoints/BuildEndpointImpl.java (hunk 3 above)
Comment (janinko) [reply rest-new/src/main/java/org/jboss/pnc/rest/endpoints/BuildEndpointImpl.java]: Yeah, I have it planned

**Pull request merged**